vmngclient/response.py

```diff
@@ -0,0 +1,161 @@
+    else:
+        _request = request
+    debug_dict = {}
+    request_debug = {
+        "method": _request.method,
+        "url": _request.url,
+        "headers": dict(_request.headers.items()),
+        "body": getattr(_request, "body", None),
+        "json": getattr(_request, "json", None),
+    }
+    debug_dict["request"] = {k: v for k, v in request_debug.items() if v is not None}
+    if response is not None:
+        response_debug = {
+            "status": response.status_code,
+            "reason": response.reason,
+            "headers": dict(response.headers.items()),
+        }
+        try:
+            json = response.json()
+            json.pop("header", None)
+            response_debug.update({"json": json})
+        except JSONDecodeError:
+            if len(response.text) <= 1024:
+                response_debug.update({"text": response.text})
+            else:
+                response_debug.update({"text(trimmed)": response.text[:128]})
+        debug_dict["response"] = response_debug
+    return pformat(debug_dict, width=80, sort_dicts=False)
+
+
+def response_history_debug(response: Optional[Response], request: Union[Request, PreparedRequest, None]) -> str:
+    """Returns human readable string containing Request-Response history contents for given response.
+
+    Args:
+        response: Response object to be debugged (note it contains an PreparedRequest object already)
+        request: optional Request object to be debugged (considered to be latest request)
+
+    When response is provided, request argument is ignored and contents of reqest.response will be returned.
+
+    Returns:
+        str
+    """
+    if response is None:
+        return response_debug(response, request)
+    response_debugs = [response_debug(resp, None) for resp in response.history]
+    response_debugs += [response_debug(response, request)]
+    return "\n".join(response_debugs)
+
+
+class JsonPayload:
+    def __init__(self, json: Any = None):
+        self.json = json
+        self.data = None
+        self.error = None
+        self.headers = None
+        if isinstance(json, dict):
+            self.data = json.get("data", None)
+            self.error = json.get("error", None)
+            self.headers = json.get("headers", None)
+
+
+class vManageResponse(Response):
+    """Extends Response object with methods specific to vManage"""
+
+    def __init__(self, response: Response):
+        self.__dict__.update(response.__dict__)
+        try:
+            self.payload = JsonPayload(response.json())
+        except JSONDecodeError:
+            self.payload = JsonPayload()
+
+    def info(self, history: bool = False) -> str:
+        """Returns human readable string containing Request-Response contents"""
+        if history:
+            return response_history_debug(self, None)
+        return response_debug(self, None)
+
+    def parse(self, cls: Type[T]) -> T:
```

Comment:
```suggestion
    def parse_dict(self, cls: Type[T]) -> T:
```
I'm just checking if we can have either one method or two methods following the intention of responsibility of each method . If possible I would to choose one method and always use DataSequence :)

Reply:
I was considering multiple options here including single parse method for both payload types.
I picked solution with explicit return types (thus 2 methods are required)
But im open to discussion - I think it is important to get it right now (as well as naming)